smartcontract/service/native/cross_chain/lock_proxy/lock_proxy.go

```diff
@@ -0,0 +1,376 @@
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[Unlock] parse from Args.TargetAssetHash to contract address format error:%s", err)
+	}
+	toAddress, err := common.AddressParseFromBytes(args.ToAddress)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[Unlock] parse from Args.ToAddress to ONTology address format error:%s", err)
+	}
+	if args.Value == 0 {
+		return utils.BYTE_TRUE, nil
+	}
+	// unlock ont or ong from current proxy contract into toAddress
+	transferInput := getTransferInput(ont.State{contract, toAddress, args.Value})
+	if _, err = native.NativeCall(assetAddress, ont.TRANSFER_NAME, transferInput); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[Unlock] NativeCall contract:%s 'transfer(%s, %s, %d)' error:%s", hex.EncodeToString(assetAddress[:]), hex.EncodeToString(contract[:]), toAddress.ToBase58(), args.Value, err)
+	}
+
+	// make sure new crossed amount is strictly less than old crossed amount and no less than the limit
+	crossedAmount, err := getAmount(native, GenCrossedAmountKey(contract, assetAddress, unlockParam.FromChainId))
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[Unlock] getCrossedAmount error:%s", err)
+	}
+	newCrossedAmount := big.NewInt(0).Sub(crossedAmount, big.NewInt(0).SetUint64(args.Value))
+	if newCrossedAmount.Cmp(crossedAmount) != -1 {
+		return utils.BYTE_FALSE, fmt.Errorf("[Unlock] new crossedAmount:%s should be less than old crossedAmount:%s", newCrossedAmount.String(), crossedAmount.String())
+	}
+	// decrease the new crossed amount by Value
+	native.CacheDB.Put(GenCrossedAmountKey(contract, assetAddress, unlockParam.FromChainId), utils.GenVarBytesStorageItem(newCrossedAmount.Bytes()).ToArray())
+
+	AddUnLockNotifications(native, contract, unlockParam.FromChainId, unlockParam.FromContractHashBs, assetAddress, toAddress, args.Value)
+
+	return utils.BYTE_TRUE, nil
+}
+
+func WithdrawONG(native *native.NativeService) ([]byte, error) {
+	contract := native.ContextRef.CurrentContext().ContractAddress
+	source := common.NewZeroCopySource(native.Input)
+	toAddress, eof := source.NextAddress()
+	if eof {
+		return utils.BYTE_FALSE, fmt.Errorf("[WithdrawONG] input DecodeAddress toAddress error!")
+	}
+	operatorAddress, err := types.AddressFromBookkeepers(genesis.GenesisBookkeepers)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[WithdrawONG] get operator error: %v", err)
+	}
+	//check witness
+	if err = utils.ValidateOwner(native, operatorAddress); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[WithdrawONG] checkWitness error: %v", err)
+	}
+	// query unbound ong or allowance
+	allowanceInput := getAllowanceInput()
+	allowanceRes, err := native.NativeCall(utils.OngContractAddress, ont.ALLOWANCE_NAME, allowanceInput)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("[WithdrawONG] invoke ong contract get allowance error:%s", err)
+	}
+	allowance := common.BigIntFromNeoBytes(allowanceRes.([]byte))
```

Comment:
suggest changing NativeCall to return []byte instead interface{}.

Reply:
updated